Add release notes 2.10.11 (#9737)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

> ## Release preview

Reviewer: ```suggestion
## Release 2.10.11
```

- Adds work pool creation wizard — https://github.com/PrefectHQ/prefect/pull/9717
- Add interactive prompts for deployment name and work pool to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/9707
- Add interactive prompt to `prefect work-pool create` — https://github.com/PrefectHQ/prefect/pull/9686

### Enhancements
- Emit events from deployments, work queues, and work pools. — https://github.com/PrefectHQ/prefect/pull/9635
- Improve SQLite database transaction behavior — https://github.com/PrefectHQ/prefect/pull/9594
- Add support for SQLAlchemy 2 — https://github.com/PrefectHQ/prefect/pull/9656
- Add `on_cancellation` flow run state change hook — https://github.com/PrefectHQ/prefect/pull/9389
- Add an `/api/ready check` endpoint to Prefect server — https://github.com/PrefectHQ/prefect/pull/9701

> - Guard against changing the profile path from `prefect config set` — https://github.com/PrefectHQ/prefect/pull/9696

Reviewer: ```suggestion
- Add guard against changing the profile path from `prefect config set` — https://github.com/PrefectHQ/prefect/pull/9696
```

- Use flow run logger to report traceback for failed submissions — https://github.com/PrefectHQ/prefect/pull/9733

> - Handle deleted deployments gracefully — https://github.com/PrefectHQ/prefect/pull/9464

Reviewer: "Fix bug where agents would fail when processing runs with deleted deployments"

more out of enhancements

- Improve default Prefect image tag when using development versions — https://github.com/PrefectHQ/prefect/pull/9503

> - Add worker event telemetry — https://github.com/PrefectHQ/prefect/pull/9702

Reviewer 2: ```suggestion
- Add worker event telemetry — https://github.com/PrefectHQ/prefect/pull/9702
- Add "Retrying" to ui states select — https://github.com/PrefectHQ/prefect-ui-library/pull/1410
- Update docs links in ui to concepts when possible — https://github.com/PrefectHQ/prefect-ui-library/pull/1351
```

### Fixes
- Display CLI deprecation warnings to STDERR instead of STDOUT — https://github.com/PrefectHQ/prefect/pull/9690

> - Fix hanging flow runs from deployments when variables exist in base scope - https://github.com/PrefectHQ/prefect/pull/9665

Reviewer: ```suggestion
- Fix hanging flow runs from deployments when variables retrieved in base scope - https://github.com/PrefectHQ/prefect/pull/9665
```

> - Fix inconsistent flow run parameters when creating a new flow run or editing a deployment — https://github.com/PrefectHQ/prefect/pull/9705

Reviewer: I think this is UI specific

> - Fix max value when updating variables — https://github.com/PrefectHQ/prefect/pull/9710

Reviewer: Max value of what?

- Allow table artifacts `table` argument as list of lists — https://github.com/PrefectHQ/prefect/pull/9732

> - Use `PrefectHttpxClient` in cloud communication — https://github.com/PrefectHQ/prefect/pull/9724

Reviewer: We can move to enhancements — would rephrase as "Use base client for Prefect Cloud communication to include automatic retries"

Reviewer: or "Add request retry support to Prefect Cloud client"

> - Remove trailing slashes when building events/in websocket url — https://github.com/PrefectHQ/prefect/pull/9663

Reviewer 2: ```suggestion
- Remove trailing slashes when building events/in websocket url — https://github.com/PrefectHQ/prefect/pull/9663
- Update flow run logs empty text for light mode
```

> - Remove trailing slashes when building events/in websocket url — https://github.com/PrefectHQ/prefect/pull/9663

Reviewer: ```suggestion
- Fix bug where events worker would fail if the API URL including a trailing `/` — https://github.com/PrefectHQ/prefect/pull/9663
```

### Deprecations

> - Rename `docker` modules to avoid conflict — https://github.com/PrefectHQ/prefect/pull/8788

Reviewer: Can you pull the specific rename notes from that pull request?

- Add worker event telemetry — https://github.com/PrefectHQ/prefect/pull/9702
- Add "Retrying" to ui states select — https://github.com/PrefectHQ/prefect-ui-library/pull/1410
- Update docs links in ui to concepts when possible — https://github.com/PrefectHQ/prefect-ui-library/pull/1351

### Fixes
- Display CLI deprecation warnings to STDERR instead of STDOUT — https://github.com/PrefectHQ/prefect/pull/9690
- Fix hanging flow runs from deployments when variables exist in base scope - https://github.com/PrefectHQ/prefect/pull/9665
- Fix inconsistent flow run parameters when creating a new flow run or editing a deployment — https://github.com/PrefectHQ/prefect/pull/9705
- Fix max value when updating variables — https://github.com/PrefectHQ/prefect/pull/9710
- Allow table artifacts `table` argument as list of lists — https://github.com/PrefectHQ/prefect/pull/9732
- Use `PrefectHttpxClient` in cloud communication — https://github.com/PrefectHQ/prefect/pull/9724
- Remove trailing slashes when building events/in websocket url — https://github.com/PrefectHQ/prefect/pull/9663

> - Update flow run logs empty text for light mode

Reviewer 2: ```suggestion
- Unknown states crash flow run timeline — https://github.com/PrefectHQ/prefect-ui-library/pull/1448
```

# Prefect Release Notes

## Release 2.10.11

- Adds work pool creation wizard — https://github.com/PrefectHQ/prefect/pull/9717
- Add interactive prompts for deployment name and work pool to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/9707
- Add interactive prompt to `prefect work-pool create` — https://github.com/PrefectHQ/prefect/pull/9686

### Enhancements

> - Emit events from deployments, work queues, and work pools. — https://github.com/PrefectHQ/prefect/pull/9635

Reviewer 3: ```suggestion
- Emit events from deployments, work queues, and work pools — https://github.com/PrefectHQ/prefect/pull/9635
```

- Add support for SQLAlchemy 2 — https://github.com/PrefectHQ/prefect/pull/9656
- Add `on_cancellation` flow run state change hook — https://github.com/PrefectHQ/prefect/pull/9389
- Improve cancellation cleanup service iteration over subflow runs - https://github.com/PrefectHQ/prefect/pull/9731
- Add `PREFECT_CLIENT_MAX_RETRIES` for configuration of maximum HTTP request retries - https://github.com/PrefectHQ/prefect/pull/9735 
- Add an `/api/ready` endpoint to the Prefect server to check database connectivity — https://github.com/PrefectHQ/prefect/pull/9701
- Display URL to flow run on creation - https://github.com/PrefectHQ/prefect/pull/9740 
- Add guard against changing the profile path from `prefect config set` — https://github.com/PrefectHQ/prefect/pull/9696
- Use flow run logger to report traceback for failed submissions — https://github.com/PrefectHQ/prefect/pull/9733
- Improve default Prefect image tag when using development versions — https://github.com/PrefectHQ/prefect/pull/9503
- Emit worker event when a flow run is scheduled to run or cancel — https://github.com/PrefectHQ/prefect/pull/9702

> - Add "Retrying" to ui states select — https://github.com/PrefectHQ/prefect-ui-library/pull/1410

Reviewer: ? What does this mean to a user?

> - Update docs links in ui to concepts when possible — https://github.com/PrefectHQ/prefect-ui-library/pull/1351

Reviewer: Does this belong in the docs section?
```suggestion
- Update docs links in UI to concepts when possible — https://github.com/PrefectHQ/prefect-ui-library/pull/1351
```

Author: I think that makes sense

### Fixes
- Display CLI deprecation warnings to STDERR instead of STDOUT — https://github.com/PrefectHQ/prefect/pull/9690
- Fix hanging flow runs from deployments when variables retrieved in base scope - https://github.com/PrefectHQ/prefect/pull/9665
- Fix maximum character length when updating variables — https://github.com/PrefectHQ/prefect/pull/9710
- Fix bug where agents would fail when processing runs with deleted deployments — https://github.com/PrefectHQ/prefect/pull/9464
- Fix bug where `uvicorn` could not be found when server was started from an unloaded virtual environment - https://github.com/PrefectHQ/prefect/pull/9734 
- Allow table artifacts `table` argument as list of lists — https://github.com/PrefectHQ/prefect/pull/9732

> - Add request retry support to Prefect Cloud client — https://github.com/PrefectHQ/prefect/pull/9724

Reviewer: This is an enhancement

- Fix bug where events worker would fail if the API URL includes a trailing `/` — https://github.com/PrefectHQ/prefect/pull/9663

> - Unknown states crash flow run timeline — https://github.com/PrefectHQ/prefect-ui-library/pull/1448

Reviewer: ?

Author: Yes 👍

- Add `on_cancellation` flow run state change hook — https://github.com/PrefectHQ/prefect/pull/9389
- Improve cancellation cleanup service iteration over subflow runs - https://github.com/PrefectHQ/prefect/pull/9731
- Add request retry support to Prefect Cloud client — https://github.com/PrefectHQ/prefect/pull/9724
- Add `PREFECT_CLIENT_MAX_RETRIES` for configuration of maximum HTTP request retries - https://github.com/PrefectHQ/prefect/pull/9735 
- Add an `/api/ready` endpoint to the Prefect server to check database connectivity — https://github.com/PrefectHQ/prefect/pull/9701
- Display URL to flow run on creation - https://github.com/PrefectHQ/prefect/pull/9740 
- Add guard against changing the profile path from `prefect config set` — https://github.com/PrefectHQ/prefect/pull/9696
- Use flow run logger to report traceback for failed submissions — https://github.com/PrefectHQ/prefect/pull/9733
- Improve default Prefect image tag when using development versions — https://github.com/PrefectHQ/prefect/pull/9503
- Emit worker event when a flow run is scheduled to run or cancel — https://github.com/PrefectHQ/prefect/pull/9702

> - Add ability in the Prefect UI to filter Task Runs tab for `Retrying` state — https://github.com/PrefectHQ/prefect-ui-library/pull/1410

Reviewer 4: ```suggestion
- Add ability to filter for `Retrying` state in the Task Runs tab of the Prefect UI — https://github.com/PrefectHQ/prefect-ui-library/pull/1410
```

## Release preview

- Adds work pool creation wizard — https://github.com/PrefectHQ/prefect/pull/9717
- Add interactive prompts for deployment name and work pool to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/9707
- Add interactive prompt to `prefect work-pool create` — https://github.com/PrefectHQ/prefect/pull/9686

### Enhancements
- Emit events from deployments, work queues, and work pools. — https://github.com/PrefectHQ/prefect/pull/9635
- Improve SQLite database transaction behavior — https://github.com/PrefectHQ/prefect/pull/9594
- Add support for SQLAlchemy 2 — https://github.com/PrefectHQ/prefect/pull/9656
- Add `on_cancellation` flow run state change hook — https://github.com/PrefectHQ/prefect/pull/9389

> - Add an `/api/ready check` endpoint to Prefect server — https://github.com/PrefectHQ/prefect/pull/9701

Reviewer: ```suggestion
- Add an `/api/ready` endpoint to the Prefect server to check database connectivity — https://github.com/PrefectHQ/prefect/pull/9701
```

- Guard against changing the profile path from `prefect config set` — https://github.com/PrefectHQ/prefect/pull/9696
- Use flow run logger to report traceback for failed submissions — https://github.com/PrefectHQ/prefect/pull/9733
- Handle deleted deployments gracefully — https://github.com/PrefectHQ/prefect/pull/9464
- Improve default Prefect image tag when using development versions — https://github.com/PrefectHQ/prefect/pull/9503

> - Add worker event telemetry — https://github.com/PrefectHQ/prefect/pull/9702

Reviewer: "telemetry" is a bit of a loaded term — can we talk about the kind of worker events we are emitting now?
